### PR TITLE
Remove data variant and data singleton. Add logging for OppositeDirec…

### PR DESCRIPTION
**Pull request by lchojnack**
Changes: 
- Remove data variant and data singleton
- Add logging for OppositeDirection situation.

Merge together with PR https://github.com/tier4/rss_evaluation_tools/pull/119

#### Description

Fixes #  

#### Where has this been tested?

  * **Platform(s):** ...
  * **Library version:** ...

### CODE REVIEW

File: ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp [hunk 1]
```diff
@@ -109,7 +110,8 @@ bool RssStructuredSceneIntersectionChecker::checkIntersectionSafe(Situation cons
   bool result = true;
   isSafe = false;
   const double not_calculated = -1.0;
-  auto &data_intersection = logging::DataIntersection::getInstance();
+  auto &extended_situation_data = logging::ExtendedSituationData::getInstance();
+  auto &data_intersection = extended_situation_data.situation_data.back().data_intersection_.value();
```
Comment (adamkrawczyk) [inline ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp]: Maybe do some func which could also check the safety?

**Review by adamkrawczyk — CHANGES_REQUESTED** [review]

File: ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp [hunk 2]
```diff
@@ -25,7 +25,9 @@ using situation::calculateTimeToCoverDistance;
 bool RssStructuredSceneIntersectionChecker::checkLateralIntersect(Situation const &situation, bool &isSafe)
 {
   isSafe = false;
-  auto &data_intersection = logging::DataIntersection::getInstance();
+  auto &extended_situation_data = logging::ExtendedSituationData::getInstance();
+  auto &data_intersection = extended_situation_data.safeGetLastSituationDataElement().getDataIntersection();
+  // auto &data_intersection = extended_situation_data.situation_data.back().data_intersection_.value();
```
Comment (lchojnack) [inline ad_rss/src/situation/RssStructuredSceneIntersectionChecker.cpp]: Remove commented code

File: ad_rss/src/logging/ExtendedSituationData.cpp [hunk 3]
```diff
@@ -30,21 +30,55 @@ void ExtendedSituationData::setSituationData(SituationData &situation)
   this->situation_data.push_back(situation);
 }
 
-// SituationData
+SituationData & ExtendedSituationData::safeGetLastSituationDataElement(){
```
Comment (lchojnack) [inline ad_rss/src/logging/ExtendedSituationData.cpp]: Lintting issue.  '{' In new line

File: ad_rss/src/logging/ExtendedSituationData.cpp [hunk 4]
```diff
@@ -30,21 +30,55 @@ void ExtendedSituationData::setSituationData(SituationData &situation)
...
+  if (situation_data.empty()){
```
Comment (lchojnack) [inline ad_rss/src/logging/ExtendedSituationData.cpp]: Lintting issue.  '{' In new line

File: ad_rss/src/logging/ExtendedSituationData.cpp [hunk 5]
```diff
@@ -30,21 +30,55 @@ void ExtendedSituationData::setSituationData(SituationData &situation)
...
+  }
+  return situation_data.back();
+}
 
-void SituationData::setSituationData(DataIntersection &data_variant)
-{
-  data_variant_ = &data_variant;
+
+DataIntersection & SituationData::getDataIntersection(){
+  if (data_intersection_.has_value()){
```
Comment (lchojnack) [inline ad_rss/src/logging/ExtendedSituationData.cpp]: Lintting issue.  '{' In new line

File: ad_rss/src/logging/ExtendedSituationData.cpp [hunk 6]
```diff
@@ -30,21 +30,55 @@ void ExtendedSituationData::setSituationData(SituationData &situation)
...
+    return data_intersection_.value();
+  }
+  else{
```
Comment (lchojnack) [inline ad_rss/src/logging/ExtendedSituationData.cpp]: Lintting issue.  '{' In new line

File: ad_rss/src/logging/ExtendedSituationData.cpp (hunk 3 above)
Comment (adamkrawczyk) [reply ad_rss/src/logging/ExtendedSituationData.cpp]: I will format with next commit

**Pull request merged**